detect_secrets_server/storage/core/git.py

```diff
@@ -93,13 +93,30 @@ def get_baseline_file(directory, filename):
 def get_diff(directory, last_commit_hash):
     """Returns the git diff between last commit hash, and HEAD."""
     kwargs = {'should_strip_output': False}
-    return _git(
+    git_args = [
         directory,
         'diff',
         last_commit_hash,
         'HEAD',
-        '--',
-        *_filter_filenames_from_diff(directory, last_commit_hash),
+    ]
+    filenames_to_include_in_diff = _filter_filenames_from_diff(directory, last_commit_hash)
+    if (
+        filenames_to_include_in_diff
+        and
+        # This happens when there are too many files in the diff
+        # e.g. 'warning: inexact rename detection was skipped due to too many files.'
+        # We choose to not include specific files out in this case
+        # Because if we did, we would later get a
+        # `OSError: [Errno 7] Argument list too long`
+        # when including the list of files.
+        not filenames_to_include_in_diff[-1].startswith('warning:')
```

Comment:
What about `endswith('too many files.')`? https://github.com/git/git/blob/master/diff.c#L6176-L6180 I think it's more clear about what warning we care about.

Alternatively, we could just hardcode a limit for `len(filenames_to_include_in_diff)`.

Reply:
When I was testing, `[-1]` was `'warning: you may want to set your diff.renameLimit variable to at least 22199 and retry the command.'`, I'm okay with any `warning`s triggering this, as it indicates something went wrong grabbing the filenames.

Reply:
The limit is OS dependent, so doing operations to try to get the number seems more obtuse here.